Widgets_CDAT/textStyleEditorWidget/TextStyleEditor.py

```diff
@@ -0,0 +1,253 @@
+        save_cancel_row.addWidget(save_button)
+        save_cancel_row.insertStretch(1, 1)
+
+        # Set up wrap
+        wrap.addWidget(self.preview)
+        wrap.addLayout(align_angle_row)
+        wrap.addLayout(font_size_row)
+        wrap.addLayout(save_cancel_row)
+        self.setLayout(wrap)
+
+
+    def setTextObject(self, textObject):
+        self.textObject = textObject
+        self.preview.setTextObject(self.textObject)
+        self.setWindowTitle("Edit Style \"" + self.textObject.name + "\"")
+
+        # set initial values
+        cur_valign = self.textObject.valign
+        for button in self.VAGroup.buttons():
+            if cur_valign == 0 and button.text() == "Top":
+                button.setChecked(True)
+            elif cur_valign == 2 and button.text() == "Mid":
+                button.setChecked(True)
+            elif cur_valign == 4 and button.text() == "Bot":
+                button.setChecked(True)
+
+        cur_halign = self.textObject.halign
+        for button in self.HAGroup.buttons():
+            if cur_halign == 0 and button.text() == "Left":
+                button.setChecked(True)
+            elif cur_halign == 1 and button.text() == "Center":
+                button.setChecked(True)
+            elif cur_halign == 2 and button.text() == "Right":
+                button.setChecked(True)
+
+        self.angle_slider.setSliderPosition(self.textObject.angle)
+
+        self.size_box.setValue(self.textObject.height)
+
+
+    def updateButton(self, button):
+        if button.text() == "Top":
+            self.textObject.valign = "top"
+
+        elif button.text() == "Mid":
+            self.textObject.valign = "half"
+
+        elif button.text() == "Bot":
+            self.textObject.valign = "bottom"
+
+        elif button.text() == "Left":
+            self.textObject.halign = "left"
+
+        elif button.text() == "Center":
+            self.textObject.halign = "center"
+
+        elif button.text() == "Right":
+            self.textObject.halign = "right"
+
+        self.preview.setTextObject(self.textObject)
+
+
+    def updateAngle(self, angle):
+
+        self.textObject.angle = angle%360
+
+        self.preview.setTextObject(self.textObject)
+
+
+    def updateFont(self, font):
+
+        self.textObject.font = str(font)
+
+        self.preview.setTextObject(self.textObject)
+
+
+    def updateSize(self, size):
+
+        self.textObject.height = size
+
+        self.preview.setTextObject(self.textObject)
+
+    def saveAs(self):
+
+        self.win = QtGui.QInputDialog()
+
+        self.win.setLabelText("Enter New Name:")
+        self.win.accepted.connect(self.save)
+
+        self.win.show()
+        self.win.raise_()
+
+
+    def save(self):
+
+        try:
+            name = self.win.textValue()
+            self.win.close()
+        except:
+            name = self.textObject.name
+
+        self.savePressed.emit(name)
+        self.close()
+
+
+
+
+
+
+
+
+
+
+
```

Comment:
Ahem.

![4334825](https://cloud.githubusercontent.com/assets/718512/13093308/7deeb4d2-d4b9-11e5-8c76-9425fdf3b0ec.jpg)


Maybe delete a few of these extra lines :wink: